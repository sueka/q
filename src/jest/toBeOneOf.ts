import { dim, red, green } from 'chalk'
import { stringify } from 'jest-matcher-utils'

// export {}

declare global {
  namespace jest {
    interface Matchers<R> {
      toBeOneOf(...actual: any[]): R;
    }
  }
}

expect.extend({
  toBeOneOf(received, ...actual) {
    const pass = actual.includes(received)

    return {
      pass,
      message() {
        return `
${ dim('expect(') }${ red('received') }${ dim(')') }.toBeOneOf${ dim('(') }${ green('expected') }${ dim(') // includes') }

Expected array: ${ green(stringify(actual)) }
Received value: ${ red(stringify(received)) }
        `.trim()
      },
    }
  },
})
