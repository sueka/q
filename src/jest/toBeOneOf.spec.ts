import './toBeOneOf'

describe('Matchers#toBeOneOf()', () => {
  it('has a neat error message', () => {
    expect(() => expect(1).toBeOneOf([2, 3, 5, 7])).toThrowErrorMatchingSnapshot()
  })
})
