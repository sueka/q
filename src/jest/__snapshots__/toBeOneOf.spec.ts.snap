// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matchers#toBeOneOf() has a neat error message 1`] = `
"[2mexpect([22m[31mreceived[39m[2m)[22m.toBeOneOf[2m([22m[32mexpected[39m[2m) // includes[22m

Expected array: [32m[[2, 3, 5, 7]][39m
Received value: [31m1[39m"
`;
